library/x509_csr.c

```diff
@@ -100,6 +102,9 @@ static int x509_csr_parse_extensions(mbedtls_x509_csr *csr,
         extn_oid.p = *p;
         *p += extn_oid.len;
 
+        /* Get and ignore optional critical flag */
+        (void)mbedtls_asn1_get_bool(p, end_ext_data, &critical);
```

Comment:
I'm not happy with us just ignoring critical extensions that we don't support as I don't think that is correct for the spec. This wouldn't be too difficult to fix - stash the value of critical here (but default to false), then in the switch just down from here where we would normally skip an unsupported extension, do something like:
```
return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_X509_INVALID_EXTENSIONS,
                                         MBEDTLS_ERR_ASN1_INVALID_DATA);
```
I'm not sure if this is the correct error here, but you hopefully get the idea.

Reply:
Handling is now done similarly to x509_get_crt_ext.